examples/karpenter/main.tf

```diff
@@ -127,6 +147,37 @@ module "aws-eks-accelerator-for-terraform" {
   # EKS CONTROL PLANE VARIABLES
   cluster_version = local.cluster_version
 
+  # Extend cluster security group rule to allow Karpenter service to communicate with Cluster API
```

Comment:
feel free to double check me, but I believe we only need the node security group additional rule allowing the control plane to reach the node on 8443 to trigger the Karpenter webhook https://github.com/terraform-aws-modules/terraform-aws-eks/blob/3ff17205a4ead51cca993547ef3de42cc080043b/examples/karpenter/main.tf#L32-L42

Reply:
Thanks I will update this one

Reply:
Just verified and we only need one ingress rule for node security group

Reply:
@bryantbiggs It would be nice to add this rule as a default for upstream module. I think its secured since its only allowing 8443 cluster sec group

Reply:
yes, @Zvikan and I were discussing this as well. I have to think about how best to support or at least show users what ports are required for common services (ALB ingress controller needs 9443/TCP, metrics server needs 10250/tcp and 8443/tcp, etc.)

Reply:
Yeah, it's not an upstream modules responsibility. It should be done by the Add-on deployment itself. Add-ons should add the necessary rules when its deployed. 

I think we should move this rule from example to the actual add-on itself for better customer experience. We can always output the cluster and worker sec groups from the `eks-blueprints` module and pass these to Kuberentes Add-ons as input. I will not do this change in this PR but we can look at this in our next improvement.